### PR TITLE
P20-12 i18n docummentation content

### PR DESCRIPTION
**Pull request by mgc1194**
**What:** This PR adds documentation of applab, gamelab and webal to the sync. 
**Why:** Part of the documentation was already made translatable in a [previous PR](https://github.com/code-dot-org/code-dot-org/pull/47511) but the actual content regarding programming expressions was not translatable yet.
**How:** Each programming environment is sync-in, from the database to individual files located in `i18n/source/docs`. This maintains modularity of the programming environments and allows different redaction/restoration approaches depending on the environment. 
During the sync out, redacted files are restored and merged into a single file `programming_environments.{locale}.json` that is distributed for each locale.
In the model `programming_expressions.rb` the recurrent scope needed to localize the content is served through `expression_scope` and a function `get_localized_property` optimizes the calls to the `I18n` function.

Both `sync-in` and `programming_expressions` model incorporate nested code according to the structure of the database.

**Note:** This PR substitutes [PR #47868](https://github.com/code-dot-org/code-dot-org/pull/47868)

## Links

- jira ticket: [P20-12](https://codedotorg.atlassian.net/browse/P20-12)/[FND-2060](https://codedotorg.atlassian.net/browse/FND-2060)
- tested links: `http://localhost-studio.code.org:3000/docs/ide/applab/expressions/button`

## Testing story

Sync ran successfully. Screenshots include examples of newly added content.

Screenshot 2022-12-06 at 23 34 56

For languages with no available translation, the webpage still renders english content.

Screenshot 2022-12-06 at 23 35 56

## Deployment strategy

Merge with staging after HoC restricted deployment.

## Follow-up work

- Javalab and Spritelab need to be added to the sync.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/i18n/sync-in.rb [hunk 1]
```diff
@@ -127,7 +127,7 @@ def localize_docs
       )
 
       ### localize Programming Expressions
-      # expression_docs.properties["syntax"] is not translated as it is the JavaScript exprerssion syntax
+      # expression_docs.properties["syntax"] is not translated as it is the JavaScript expression syntax
```
Comment (mgc1194) [inline bin/i18n/sync-in.rb:130]: I'm really thankful for these spelling checks. Double letters throw me off all the time

File: bin/i18n/sync-in.rb [hunk 2]
```diff
@@ -87,27 +88,95 @@ def localize_standards
...
+      ### localize Programming Expressions
+      # expression_docs.properties["syntax"] is not translated as it is the JavaScript expression syntax
+      expressions_data = categories_data[category_key]["expressions"]
+      category_for_navigation[:docs].each do |expression|
+        expression_key = expression[:key]
+        expression_docs = ProgrammingExpression.find_by_id(expression[:id])
+        expressions_data.store(
+          expression_key, {
+            'content' => expression_docs.properties["content"],
+            'examples' => ({} unless expression_docs.properties['examples'].nil?),
```
Comment (annaxuphoto) [inline bin/i18n/sync-in.rb]: is there a reason 'examples' is single quoted here and on line 150?

File: bin/i18n/sync-in.rb [hunk 3]
```diff
@@ -87,27 +88,95 @@ def localize_standards
...
+            'palette_params' => ({} unless expression_docs.properties["palette_params"].nil?),
```
Comment (annaxuphoto) [inline bin/i18n/sync-in.rb:139]: trying to understand what palette_params does - where does it get set if here it's an empty hash?

**Review by annaxuphoto — COMMENTED** [review]
I have a couple small questions, but overall this looks really really good - thanks for pushing through on all of this work!!

File: bin/i18n/sync-in.rb (hunk 3 above)
Comment (mgc1194) [reply bin/i18n/sync-in.rb:139]: palette_params are the parameters of the programming expressions. Some expressions have parameters like [`button(id, text)`](https://studio.code.org/docs/ide/applab/expressions/button) where the parameters are `id` and `text`. Other expressions do not have parameters like [`buzzer`](https://studio.code.org/docs/ide/applab/expressions/buzzer) 

- In the first case, examples and parameters get set right after on line 161.

- In these second case, `expression_docs[palette_params]` returns a nil value.

File: bin/i18n/sync-in.rb (hunk 2 above)
Comment (mgc1194) [reply bin/i18n/sync-in.rb]: No reason, probably I copy-pasted with single quotes from somewhere else. I'll fix those.

Comment (mgc1194) [reply bin/i18n/sync-in.rb]: Fixed!!

**Review by annaxuphoto — APPROVED** [review]

**Pull request merged**